Doc/library/typing.rst

```diff
@@ -660,7 +667,8 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
    Optional type.
 
-   ``Optional[X]`` is equivalent to ``X | None`` (or ``Union[X, None]``).
+   ``Optional[X]`` is logically equivalent to ``X | None`` and ``Union[X, None]``.
```

Comment:
Minor suggestion:
```
``X | None`` is logically equivalent to ``Optional[X]`` and ``Union[X, None]``.

Reply:
I prefer "semantically equivalent" to "logically equivalent" 😉

Reply:
Even better.